cmd/web-handlers.go

```diff
@@ -546,10 +546,15 @@ func (web *webAPIHandlers) Download(w http.ResponseWriter, r *http.Request) {
 	objectLock.RLock()
 	defer objectLock.RUnlock()
 
-	if err := objectAPI.GetObject(bucket, object, 0, -1, w); err != nil {
+	info, err := objectAPI.GetObjectInfo(bucket, object)
```

Comment:
A simple GET Object via the S3 API is served by making object layer calls to [GetObjectInfo followed by GetObject](https://github.com/minio/minio/blob/master/cmd/object-handlers.go#L101) - it is not a single call. I think it is acceptable to do the same here. These calls are happening within the backend server and not being sent from browser to backend server.